torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -382,11 +437,23 @@ void ProfilingGraphExecutorImpl::runProfilingOptimizations(
     GRAPH_DEBUG("After CreateAutodiffSubgraphs\n", *copy);
     size_t idx = 0;
     for (Node* dnode : diff_nodes) {
-      GRAPH_DEBUG("Optimizing diff node ", idx);
+      GRAPH_DEBUG("Optimizing diff node ", idx, " in ", *copy);
+      if (!guardDifferentiableGraph(dnode)) {
+        // if we cannot guard (because of inputs without profiling information),
+        // we re-inline the subgraph and remove the differentiable node
+        GRAPH_DEBUG("Could not guardDifferentiableGraph ", idx, " in ", *copy);
+        idx++;
+        continue;
+      }
+      GRAPH_DEBUG("After guardDifferentiableGraph:\n", *copy);
       auto diff_graph = std::move(dnode->g(attr::Subgraph));
       Gradient gradient = differentiate(diff_graph);
```

Comment:
Once we re-profile within differentiable graphs, we can remove the profile nodes before differentiating, and we not compute unnecesssary gradients. I'm not sure we're doing that now.  

see: 
https://github.com/pytorch/pytorch/blob/22a34bcf4e5eaa348f0117c414c3dd760ec64b13/torch/csrc/jit/runtime/autodiff.cpp#L396